docs/user-guide/bandwidth-resources.md

```diff
@@ -0,0 +1,171 @@
+resource limits.  A *pod resource limit* for a particular resource type is the sum of the resource
+limits of that type for each container in the pod, with unset values treated as zero.
+
+The following pod has two containers.  Each has a limit of 1Mbit of ingress bandwidth and 2Mbit of
+egress bandwidth.
+The pod can be said to have a limit of 2Mbit of ingress and 4Mb of egress.
+
+*Note* Because of limitations in container runtimes, we can't actually limit the bandwidth between
+containers, so different containers in a Pod may steal from each other.  We can (and do) limit
+the overall bandwidth for the Pod.  We hope to refine this as capabilities for hierarchical
+network namespaces come into the container runtimes like Docker.
+
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: frontend
+spec:
+  containers:
+  - name: db
+    image: mysql
+    resources:
+      limits:
+        ingress: "1Mi"
+        egress: "2Mi"
+  - name: wp
+    image: wordpress
+    resources:
+      limits:
+        ingress: "1Mi"
+        egress: "2Mi"
+```
+
+## How Pods with Resource Limits are Scheduled
+
+When a pod is created, the Kubernetes scheduler selects a node for the pod to
+run on.  Each node has a maximum capacity for each of the resource types: the
+amount of CPU and memory it can provide for pods.  The scheduler ensures that,
+for each bandwidth resource type (ingress/egress), the sum of the resource limits of the
+containers scheduled to the node is less than the capacity of the node.  Note
+that although actual bandwidth resource usage on nodes is very low, the
+scheduler will still refuse to place pods onto nodes if the capacity check
+fails.  This protects against a resource shortage on a node when resource usage
+later increases, such as due to a daily peak in request rate.
+
+Note: Although the scheduler normally spreads pods out across nodes, there are currently some cases
+where pods with no limits (unset values) might all land on the same node.
+
+## How Pods with Resource Limits are Run
+
+When kubelet starts a container of a pod, it manages the network resource consumption based
+on the pod's IP address, using the `tc` command.
+
+- The `spec.container[].resources.limits.ingress` or `egress` is summed for all containers in the pod.
+- The `tc` command is used to create a bandwidth limit classes for ingress and egress. Different classes are used for each..
+- The `tc` command is used to apply this bandwidth limit class to all network traffic either originating (egress) or arriving (ingress) to the Pod's IP address.
+
+`tc` is configured to use the hierarchical token bucket (`htb`) algorithm for bandwidth shaping.  Packets are slowed down in the kernel's queues in
+order to achieve the desired bandwidth restrictions.
+
+## Monitoring Bandwidth Resource Usage
+
+It is not currently possible to monitor the bandwidth usage of a pod.  Support for this is planned.
```

Comment:
Cadvisor already provides networking data for the veth device assigned to a pod. It is currently not exposed via kubelet though.